src/Paths/FileMetadata.ts

```diff
@@ -0,0 +1,16 @@
+export class FileMetadata {
+  constructor(readonly timeOfAccess: number, readonly timeOfModification: number, readonly timeOfBirth: number) {}
+
+  static new(): FileMetadata {
+    const now = Date.now();
+    return new FileMetadata(now, now, now);
+  }
+
+  read(): FileMetadata {
+    return new FileMetadata(Date.now(), this.timeOfModification, this.timeOfBirth);
```

Comment:
I *suspect* we're going to want to update the metadata in-place, although I suppose either way can work depending on programming style.